Test example bash snippets

Twey opened the pull request:
## Motivation

We would like to ensure that bash snippets in our example READMEs are not broken.

<!-- Short text indicating what this PR aims to accomplish. -->

## Proposal

<!-- What are the proposed changes and why are they appropriate? -->

Add tests that execute the bash in the examples.

## Test Plan

<!-- How to test that the changes are correct. -->

Run `cargo test -p linera-service`.

## Release Plan

<!--
How to safely release the changes.

Please only include the relevant items (if any) and create issues to track future release work.
-->

These changes only affect testing and should be safe to release.

## Links

<!--
Optional section for related PRs, related issues, and other references.

If needed, please create issues to track future improvements and link them here.
-->
- [reviewer checklist](https://github.com/linera-io/linera-protocol/blob/main/CONTRIBUTING.md#reviewer-checklist)
- https://github.com/linera-io/linera-protocol/issues/1180
- https://github.com/linera-io/linera-protocol/issues/1176
- https://github.com/linera-io/linera-protocol/issues/1177

ma2bd commented:
Need to update Cargo.lock

ma2bd commented:
> Extract the owner and chain IDs programmatically rather than hard-coding them into the script and relying on a deterministic RNG seed.

FYI chain IDs are actually always deterministic (based on block heights, transaction index, etc)

Twey commented:
> Need to update Cargo.lock

Yes sorry, I pushed the commit for debugging purposes last night and then went to bed while the CI did its thing :)  I was trying to keep the `Cargo.lock` as similar to `main` as possible.

This commit is just for debugging purposes (generally commits labelled `XXX` are intended to not make it into the final merge) but this response from the CI is quite interesting:

Thinking that perhaps I'd accidentally pulled in a different version of `syn` when updating `test-case` I added a debugging commit that reverts that bump but does nothing else — so all `Cargo.toml` and `Cargo.lock` files should be in exactly the same state as on `main` with the singular exception that `test-case`.  But in fact on some inspection of the `syn` source code the symbols that `check-all-features` is complaining about have always been gated behind `full` back to their introduction, so it can't be a version issue.

The other errors around unused imports seem to be symptomatic of this error.

> FYI chain IDs are actually always deterministic (based on block heights, transaction index, etc)

On some consideration I think this is an invalid complaint anyway — insofar as the documentation is supposed to reflect a reasonable ‘interactive’ session, the user would probably just copy/paste the IDs out of the `linera wallet show` output.  I've removed it :)

Twey commented:
I've pushed a commit that fixes the CI — but I'm not sure that it really belongs in this pull request.  What do we think?

ma2bd commented:
> I've pushed a commit that fixes the CI — but I'm not sure that it really belongs in this pull request. What do we think?

It must somehow belong here because CI is passing on the `main` branch.

ma2bd commented on Cargo.lock:
```diff
@@ -716,15 +717,15 @@ checksum = "9e1b586273c5702936fe7b7d6896644d8be71e6314cfe09d3167c95f712589e8"
 [[package]]
 name = "bcs"
-version = "0.1.5"
+version = "0.1.6"
```
It looks like you run `cargo update`?

Twey replied:
For the fix commit yes, and also probably in the build after updating `test-case`

Twey replied:
For the final version I've removed the `test-case` update from here, preferring to move it to https://github.com/linera-io/linera-protocol/pull/1178 with others of its kind.  In order to add `test-case` to `linera-service` without doing a full `cargo update`, I manually edited `Cargo.lock` to add only the dependency we need to build without updating `Cargo.lock`.

ma2bd reviewed (approved):
LGTM

ma2bd commented:
> The two application publishes are racy — ..
> The two node service spawns are racy — ..

I think I would add a small `sleep` in the bash script to hint that something is going on. And/or add a TODO with a reference to a github issue.

Twey commented on examples/social/src/lib.rs:
````diff
@@ -1,149 +1,155 @@
-//!   }
-//! }
-//! ```
+/*!
````
I tend to use block comments for large blocks of documentation since it makes it easier to edit — if it's against Linera policy let me know and I can switch it back :)

Twey commented on examples/crowd-funding/README.md:
```diff
@@ -101,40 +93,48 @@ A table will be shown with the chains registered in the wallet and their meta-da
+
+## Creating tokens
+
+Compile the Wasm binaries for the two applications `fungible` and
```
I've refilled at 90.  Additionally I've added a `.editorconfig` file with this and other standards you've mentioned in PR https://github.com/linera-io/linera-protocol/pull/1175.

Twey commented:
>I think I would still add a small sleep in the bash script to hint that something is going on, and ideally add a TODO with a reference to a github issue.

I've created https://github.com/linera-io/linera-protocol/issues/1176 and https://github.com/linera-io/linera-protocol/issues/1177 and referenced them from comments around the documentation.

Twey commented:
I'm waiting for https://github.com/linera-io/linera-protocol/pull/1178 to merge this since once that's merged I can take out the two dependency-change PRs.
